### PR TITLE
add option to work with and keep an existing namespace

### PR DESCRIPTION
**Pull request by marnikow**
Because I manage my namespaces outside of this script an option to keep the namespace was needed so that my setup keeps on working even if the tunnel is brought down.

I added an command line option to facilitate this. Let me know if I need to adjust something or if this function doesn't fit in your view on the project.

### CODE REVIEW

**Comment by dadevel** [comment]
Hey, thanks for your contribution! It's very welcome. I have two questions:

1) What do you think about the introduction of an additional option in the `profile.json` instead of a command line flag?

    E.g.:

    ~~~ json
    {
      "name": "my-netns",
      "managed": false,
      "interfaces": [...]
    }
    ~~~

2) Does the following description match the behavior you expect?

    When running `wg-netns up` on a profile with `managed=false` the namespace would be expected to already exists, with `lo` already up and DNS already configured. Only the interfaces specified in the profile will be brought up.

    When running `wg-netns down` on a profile with `managed=false` only the interfaces specified in the profile will be deleted, but the namespace itself and all other interfaces will remain untouched.

File: wg-netns.py [hunk 1]
```diff
@@ -76,14 +79,31 @@ def profile_read(path):
...
 
+def namespace_get_list_of_existing():
+    ip_list = ip('netns', 'list', capture=True).splitlines()
+    rg = re.compile('(?P<name>[^ ]*)(?: \(id: (?P<id>\d+)\))?')
+    existing_namespaces = dict()
+    for line in ip_list:
+        match = rg.fullmatch(line)
+        if match:
+            existing_namespaces[match.group("name")] = match.group("id")
+    return existing_namespaces
```
Comment (dadevel) [inline wg-netns.py:99]:
```suggestion
    namespaces = json.loads(ip('-j', 'netns', 'list', capture=True))
    return {namespace['name'] for namespace in namespaces}
```

**Comment by marnikow** [comment]
Hi

1. Would certainly work.
2. It almost match. `lo` and DNS are not necessary already configured when running  `wg-netns up`. Aside from that, I think that the option to update DNS should be available even when managed. This allows to use different wireguard profiles at different times without adjusting DNS manually.

**Comment by dadevel** [comment]
Yeah, that make sense.  
If the `dns-server` option is present, DNS is managed by `wg-netns` otherwise not. This should be implemented already.  
And `lo` will be silently added if missing.

**Comment by marnikow** [comment]
Thanks for the work!

**Comment by dadevel** [comment]
Let me know if you find any bugs.

**Pull request closed without merging**